_keystone/k2k-federation.md

```diff
@@ -0,0 +1,582 @@
+---
+layout: page
+title: Keystone to Keystone (K2K) Federation on Blue Box
+featured: false
+tags: [federation, keystone, k2k]
+dateAdded: June 24th, 2016
+author: Elvin Tubillara
+editor:
+---
+
+* [What is K2K Federation?](#what_is_k2k?)
+* [Managing Groups](#managin_groups)
+* [Managing Mappings](#managing_mappings)
+* [Using Horizon](#using_horizon)
+* [Using the Python API Libraries](#using_the_python_api_libraries)
+* [More Mappings and Groups](#more_mappings_and_groups)
+
+## <a name="what_is_k2k"></a>What is K2K Federation?
+K2K federation is a way to use credentials from one Keystone across multiple
+Blue Box clouds. A Blue Box cloud instance is designated as an identity provider
+and other cloud instances are designated as service providers. The
+identity provider's job is to store the user's credentials and assert
+to the service providers that the user is a valid user. The user/pass
+is never sent to the service providers (only assertions from the identity provider). The user signs into
+the identity provider, gets a signed SAML assertion from the identity provider
+and uses the SAML assertion to authenticate into the service providers. The log in process
+for handling the SAML flow is handled by the K2K authentication plugin (found in keystoneauth1).
+This allows credentials to be managed only on the identity provider
+and be accepted by service providers through assertions.
+
+If the several Blue Box clouds have been set up with the K2K feature, one
```

Comment:
This sentence should be.
`If several BlueBox clouds have been setup with K2K feature, one will be designated as Identity Provider and others as Service Providers.`